scripts/drone/pipelines/benchmarks.star

```diff
@@ -0,0 +1,114 @@
+"""
+This module returns the pipeline used for integration benchmarks.
+"""
+
+load(
+    "scripts/drone/steps/lib.star",
+    "compile_build_cmd",
+    "enterprise_setup_step",
+    "verify_gen_cue_step",
+    "verify_gen_jsonnet_step",
+    "wire_install_step",
+)
+load(
+    "scripts/drone/services/services.star",
+    "integration_test_services",
+    "integration_test_services_volumes",
+)
+load(
+    "scripts/drone/utils/utils.star",
+    "pipeline",
+)
+load(
+    "scripts/drone/utils/images.star",
+    "images",
+)
+
+def integration_benchmarks(prefix):
+    """Generate a pipeline for integration tests.
+
+    Args:
+      prefix: used in the naming of the pipeline.
+    Returns:
+      Drone pipeline.
+    """
+    environment = {"EDITION": "oss"}
+
+    services = integration_test_services()
+    volumes = integration_test_services_volumes()
+
+    # In pull requests, attempt to clone grafana enterprise.
+    init_steps = [enterprise_setup_step(allowPromote = True)]
+
+    verify_step = verify_gen_cue_step()
+    verify_jsonnet_step = verify_gen_jsonnet_step()
+
+    # Ensure that verif_gen_cue happens after we clone enterprise
+    # At the time of writing this, very_gen_cue is depended on by the wire step which is what everything else depends on.
+    verify_step["depends_on"].append("clone-enterprise")
+    verify_jsonnet_step["depends_on"].append("clone-enterprise")
+
+    init_steps += [
+        compile_build_cmd(),
+        verify_step,
+        verify_jsonnet_step,
+        wire_install_step(),
+    ]
+
+    cmd = [
+        "if [ -z ${GO_PACKAGES} ]; then echo 'missing GO_PACKAGES'; false; fi",
+        "go test -v -run=^$ -benchmem -timeout=1h -count=8 -bench=. ${GO_PACKAGES}",
+    ]
+
+    benchmark_steps = [
```

Comment:
These are basically the same environments we also have for the integration tests, right? Could this perhaps be combined somehow so that there is just a loop that sets the steps up? Would looping over `services` work here?

Reply:
I don't think looping over services would help much because they do not match exactly:
- the environmental variables differ (eg `POSTGRES_PASSWORD` vs `PGPASSWORD`)
- we would need to filter out the cache services whereas we would need an additional step for `sqlite`

However, I did some refactoring to reduce some duplicated code
would you like to check once again?